app.go

```diff
@@ -88,6 +88,10 @@ type App struct {
 	// single-character bool arguments into one
 	// i.e. foobar -o -v -> foobar -ov
 	UseShortOptionHandling bool
+	// This function is automatically executed when the context is cancelled due to the program
+	// receiving an interrupt signal (Control+C). This function can be overridden to implement
+	// desired behaviour in such a scenario
+	InterruptHandler InterruptHandlerFunc
```

Comment:
The function is executed when the application receives and interuption signal, what that does is configurable based on what the handler is doing.

Reply:
If we change this to signal handler, we should change the comment quite heavily.